### PR TITLE
Adds split method to port class

### PR DESCRIPTION
**Pull request by mvillius**
Pretty self-explanatory

### CODE REVIEW

File: HFSSdrawpy/core/port.py [hunk 1]
```diff
@@ -183,3 +184,49 @@ def rotate_ports(ports, angle):
...
+
+        if not subnames:
+            subnames = self.subnames
+        elif subnames == -1:
+            subnames = self.subnames[:-1]
+        if not all(sub in self.subnames for sub in subnames):
+            raise ValueError('Split ports must belong to '+str(self.subnames))
+
+        self.save = {'name':self.name, 'pos':self.pos, 'widths':self.widths,
+                     'subnames':self.subnames, 'offsets':self.offsets}
```
Comment (rlescanne) [inline HFSSdrawpy/core/port.py]: This is not needed here (I think). The save is only required when performing a brief modification of the port (such as when drawing adaptors, the port is temporarily shifted and resized to the other port specs).

File: HFSSdrawpy/core/port.py [hunk 2]
```diff
@@ -183,3 +184,49 @@ def rotate_ports(ports, angle):
             posx = port.pos[0]*np.cos(rad)+port.pos[1]*np.sin(-rad)
             posy = port.pos[0]*np.sin(rad)+port.pos[1]*np.cos(rad)
             port.pos = Vector([posx, posy])
+
+    def split(self, subnames=[], gap=None):
```
Comment (rlescanne) [inline HFSSdrawpy/core/port.py]: Je ne sais plus pourquoi mais on n'a pas le droit d'initialiser par défaut à une liste en python. Il vaut mieux mettre un None et propager cette modification dans ton code.

File: HFSSdrawpy/core/port.py [hunk 3]
```diff
@@ -183,3 +184,49 @@ def rotate_ports(ports, angle):
...
+        Returns
+        -------
+        subports : list of Port instances
+
+        """
+
+        if not subnames:
+            subnames = self.subnames
+        elif subnames == -1:
+            subnames = self.subnames[:-1]
```
Comment (rlescanne) [inline HFSSdrawpy/core/port.py]: I guess this assumes the last subport defined is the outline/gap. This assumption is not ideal but I guess this is the best solution for now.

File: HFSSdrawpy/core/port.py [hunk 4]
```diff
@@ -183,3 +184,49 @@ def rotate_ports(ports, angle):
...
+
+        subports = []
+        for sub in subnames:
+            isub     = self.subnames.index(sub)
+            if gap is None:
+                gap = self.widths[isub]
+            name     = self.name+'_'+sub
+            pos      = self.pos + Vector([0,self.offsets[isub]]).rot(self.ori)
+            widths   = [self.widths[isub], self.widths[isub]+2*gap]
+            layers   = [TRACK, GAP]
```
Comment (rlescanne) [inline HFSSdrawpy/core/port.py]: To add a little bit of flexibility here, I would replace `TRACK` by the layer of the corresponding subport. I agree that we cannot give this flexibility to `GAP`.

**Review by rlescanne — CHANGES_REQUESTED** [review]
Qqs petites remarques

**Comment by mvillius** [comment]
Done

**Comment by mvillius** [comment]
I added the possibility to split a multiport (non-CPW) into several multiports, and since the previous Pull Request had not been closed it has been included in the existing one. What do you think Raph?

File: HFSSdrawpy/core/port.py [hunk 5]
```diff
@@ -105,9 +105,11 @@ def compare(self, other, pm, slope=0.5):
             width2 = other.widths[ii]
 
             offset1 = self.offsets[ii]
-            offset2 = other.offsets[ii]
+
+            offset2 = -other.offsets[ii]
```
Comment (pierreguilmin) [inline HFSSdrawpy/core/port.py:109]: @MVillius do you think I can merge this modification into master or is it just a specific fix for your work?

File: HFSSdrawpy/interfaces/gds_modeler.py [hunk 6]
```diff
@@ -4,7 +4,7 @@
 from ..core.entity import gen_name
 from ..utils import Vector, parse_entry, val
 
-TOLERANCE = 1e-8  # for arcs
+TOLERANCE = 1e-9 # for arcs
```
Comment (mvillius) [inline HFSSdrawpy/interfaces/gds_modeler.py:7]: This is the new standard for everybody in the lab, we should keep 1e-9

File: HFSSdrawpy/core/port.py (hunk 5 above)
Comment (mvillius) [reply HFSSdrawpy/core/port.py:109]: This modification makes your life easier when you want to draw adaptors with off-centered multiports. I think I am the only one to do this so you should be able to merge this modification into master seamlessly

**Pull request merged**